validphys2/src/validphys/checks.py

```diff
@@ -235,6 +236,18 @@ def check_fits_different(fits):
     """Need this check because oterwise the pandas object gets confused"""
     return _check_list_different(fits, 'fits')
 
+@make_argcheck
+def check_fit_versions_equal(fit):
+    """Check that all replicas in the fit used matching versions of the repository
+    and python modules"""
+    original_version = None
+    for json_path in fit.path.glob("nnfit/replica_*/*.json"):
```

Comment:
Probably the second glob should be a bit more specific (there could be other json files). Btw a pet peeve is have these variables file names that mke processing that much harder (the right name is `replica_info.json` or something). Is that too difficult to change?

Reply:
Actually, I wanted to use the `replica_paths` but was not able to make vp understand that I wanted it there?

> the right name is replica_info.json or something

You mean, instead of <fit_name>.json? If that's your suggestion I agree, I'd be happy to change it.

Reply:
The following might not be relevant here at all but in the case of proton+nuclear fits there is not only one single `json` file (since each nucleus has its own `json`) and this way of doing it is the easiest.

Reply:
Does it make sense to have different `json` files? What information is different between different nucleus?

Reply:
Only the `preprocessing` which in principle could actually **all** be dumped into a single `json` file.

Reply:
tbh I don't think the preprocessing is used? So I would not fret about it...

Reply:
Yes, this information is not used at all currently, however, in the case of nPDFs where the integrability is not enforced in the way it is done in the free-proton, knowing the effective exponents allows one the verify that a posteriori. 

But I agree that there is actually no need for different `json` files and that specifying the name is the correct way to do it.

Reply:
If you are happy with `fitinfo.json` I'll do that (with a fallback for compatibility ofc)